tlatools/org.lamport.tlatools/src/tlc2/tool/impl/DebugTool.java

```diff
@@ -123,20 +124,28 @@ public boolean isValidAssumption(final ExprNode assumption) {
 	}
 
 	@Override
-	public boolean isValid(Action act, TLCState state) {
+	public InvariantViolation findFalsifyingContext(Action invariant, TLCState s0, TLCState s1) {
+		// Need this override for breakpoints to work
```

Comment:
What exactly doesn't work without this override?

Reply:
The old `isValid` method was just a wrapper to `eval`. The new `findFalsifyingContext` method actually bypasses some calls to `eval`---and something about that causes a test hang. I know that's not a very clear explanation, but I don't fully understand the problem myself yet.